src/Tabs/TabIndicator.js

```diff
@@ -0,0 +1,54 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import bemClassNames from '../utils/bemClassNames';
+import './tabindicator.css';
+
+const bem = bemClassNames('d2ui-tab-indicator');
+
+class TabIndicator extends Component {
+    constructor(props) {
+        super(props);
+        this.state = { animated: false };
+    }
+
+    componentDidUpdate(prevProps) {
+        // The tabindicator should not move with a CSS transition when it is first positioned into place
+        // The tick after the visibility is true, the animation should start
+        if (!prevProps.visible && this.props.visible) {
+            this.setState({
+                animated: true,
+            });
```

Comment:
So effectively, this component is forced through at least three render cycles before is visible:

1. First tick the element to position itself against is not known yet
2. Second tick, the element is known so the indicator can show itself at the right spot. However, at the moment the element has no CSS transition applied to it, because we want to show it at the right place in one go.
3. Now the element has moved into its initial position, on any subsequent tab-change, we want to do an animated move, so the CSS transition is added.

Reply:
Hmmm, not at the top of my head. I'll think about it some. Do you have a good way to capture the problem in a measurable way?

Reply:
Keeping this for now as discussed